src/actors.rs

```diff
@@ -241,9 +246,13 @@ where
         let metadata = workers::Metadata::new(db_pool.clone(), ctx.meta().clone())
             .await?
             .spawn();
-        let blocks =
-            workers::BlocksIndexer::new(ctx.backend().clone(), db_pool.clone(), metadata.clone())
-                .spawn();
+        let blocks = workers::BlocksIndexer::new(
+            ctx.backend().clone(),
+            db_pool.clone(),
+            metadata.clone(),
+            ctx.max_block_load,
+        )
```

Comment:
Do you think `BlocksIndexer::new` could be changed to take only a `ctx` and `db_pool`? Seems like we're piling on arguments here and that could be done in `new()` itself?

Reply:
In order to construct metadata within `BlocksIndexer::new` we would need to make it an async function and copy the code for instantiating the metadata actor, which imo seems a bit heavy handed. I did however remove the `backend` and `max_block_load` param since they are directly on `ctx`. 

I think if I was trying to do this in TS I would pass the same args in as originally, but just use an object for the argument, which can be cleanly de-structured - I don't think there is clean equivalent in rust for de-structuring params like that though. The reason I bring it up is that I think it decouples the function from the interface of `ctx` and imo it is more declarative/ maintains the division of responsibility.

Reply:
I think sometime in the future we may try to refactor the architecture of `System` and how it spawns/creates actors and connects external services (client/backend) to them, possibly by splitting it up into two separate structs or some other strategy. It's one of the parts of archive that I've struggled  to 'get right' in terms of the API, and never have been quite satisfied with.